tools/flash_shelly.py

```diff
@@ -0,0 +1,499 @@
+  except getopt.GetoptError as err:
+    logger.error(err)
+    usage()
+    exit(2)
+
+  action = "flash"
+  do_all = False
+  dry_run = False
+  silent_run = False
+  forced_version = False
+  exclude = False
+  mode = 'homekit'
+  ffw = None
+
+  for (opt, value) in opts:
+    if opt == "-m":
+      if value == 'homekit':
+        mode = 'homekit'
+      elif value == "revert":
+        mode='stock'
+      elif value == "keep":
+        mode="keep"
+      else:
+        print("Invalid option")
+        usage()
+        exit(2)
+    elif opt == "-a":
+      do_all = True
+    elif opt == "-e":
+      exclude = True
+    elif opt == "-l":
+      action = "list"
+    elif opt == "-n":
+      dry_run = True
+    elif opt == "-y":
+      silent_run = True
+    elif opt == "-V":
+      forced_version = True
+      ffw = value
+    elif opt == '-D':
+      if logger.getEffectiveLevel() >= 30:
+        logger.setLevel(logging.INFO)
+      else:
+        logger.setLevel(logging.DEBUG)
+    elif opt == '-h':
+      usage()
+      exit(0)
+
+  logger.info(WHITE + "app" + NC)
+  logger.info(PURPLE + "OS: %s\033[0m"% arch)
+  logger.info("ARG: %s" % argv)
+  logger.info("opts: %s" % opts)
+  logger.info("args: %s" % args)
+  logger.info("action: %s" % action)
+  logger.info("do_all: %s" % do_all)
+  logger.info("dry_run: %s" % dry_run)
+  logger.info("silent_run: %s" % silent_run)
+  logger.info("forced_version: %s" % forced_version)
+  logger.info("exclude: %s" % exclude)
+  logger.info("mode: %s" % mode)
+
+  if ((not opts and not args) and do_all == False) or (not args and do_all == False) or \
+     (args and do_all == True and exclude == False) or (not args and do_all == True and exclude == True):
+    usage()
+    exit(1)
+
+  try:
+    fp = urllib.request.urlopen("https://api.shelly.cloud/files/firmware")
+    global stock_release_info
+    stock_release_info = json.load(fp)
+    fp.close()
+  except:
+    logger.warning("Failed to lookup version information")
+    exit(1)
+  try:
+    fp = urllib.request.urlopen("https://rojer.me/files/shelly/update.json")
+    global homekit_release_info
+    homekit_release_info = json.load(fp)
+    fp.close()
+  except:
+    logger.warning("Failed to lookup version information")
+    exit(1)
+
+  logger.debug("\n" + WHITE + "stock_release_info:" + NC + " %s" % stock_release_info)
+  logger.debug("\n" + WHITE + "homekit_release_info:" + NC + " %s" % homekit_release_info)
+
+  if args and exclude == False:
+    for device in args:
+      device_scan(device, action, do_all, dry_run, silent_run, mode, exclude, forced_version, ffw)
+
+  if do_all == True and exclude == False:
+    device_scan("", action, do_all, dry_run, silent_run, mode, exclude, forced_version, ffw)
+
+  if do_all == True and exclude == True:
+    device_scan(args, action, do_all, dry_run, silent_run, mode, exclude, forced_version, ffw)
+
+
+
+
+if __name__ == '__main__':
+  app(argv)
+
+__all__ = ['app']
```

Comment:
why is this needed?

Reply:
runs the script from cmd line.

Reply:
it required to pass args and opts to getopt, they are stored in the import `argv `

Reply:
ie

```
if __name__ == "__main__":
   print("File one executed when ran directly")
else:
   print("File one executed when imported")
```

Reply:
I can remove `__all__ = ['app']` now we are not using `from xxx import yyyy`

Reply:
https://github.com/mongoose-os-apps/shelly-homekit/pull/176/commits/5286c45882aa179560b71b4ab5957a26d3397d00

Reply:
https://github.com/mongoose-os-apps/shelly-homekit/pull/176/commits/ae906910935bd41b57675857f7d42251938dad4c

Reply:
https://github.com/mongoose-os-apps/shelly-homekit/pull/176/commits/f9c90340cd36225fb2190d643fb8628d2b882543

Reply:
should be resolved now.